solidity/ecdsa/test/WalletRegistry.RandomBeacon.test.ts

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable no-underscore-dangle */
+import {
+  ethers,
+  waffle,
+  helpers,
+  deployments,
+  getUnnamedAccounts,
+} from "hardhat"
+import { expect } from "chai"
+
+import { walletRegistryFixture } from "./fixtures"
+
+import type { ContractTransaction } from "ethers"
+import type { MockContract } from "ethereum-waffle"
+import type { WalletRegistry, WalletRegistryStub } from "../typechain"
+import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers"
+
+const { createSnapshot, restoreSnapshot } = helpers.snapshot
+
+const { deployMockContract } = waffle
+
+describe("WalletRegistry - Random Beacon", async () => {
+  let walletRegistry: WalletRegistryStub & WalletRegistry
+  let randomBeacon: MockContract
+
+  let deployer: SignerWithAddress
+  let walletOwner: SignerWithAddress
+  let thirdParty: SignerWithAddress
+
+  before("load test fixture", async () => {
+    // eslint-disable-next-line @typescript-eslint/no-extra-semi
+    ;({ walletRegistry, walletOwner, deployer, thirdParty } =
+      await waffle.loadFixture(walletRegistryFixture))
+
+    const randomBeaconMock = await deployMockContract(
+      deployer,
+      (
+        await deployments.getArtifact("RandomBeacon")
+      ).abi
+    )
+
+    await walletRegistry.updateRandomBeacon(randomBeaconMock.address)
```

Comment:
At the end we'll need to update a random beacon contract through governance, cause 'updateRandomBeacon()` will be ownable ie. `walletRegistryGovernance.connect(governance).updateRandomBeacon(newRandomBeacon.address)`. Can we add a TODO here?